database/buildSchema/43_views_functions_triggers.sql

```diff
@@ -1683,78 +1681,48 @@ BEGIN
             WHERE id = NEW.application_id;
     END IF;
 
-    -- Get list of reviewer_ids to update for 
-    IF TG_OP = 'UPDATE' THEN
-        reviewer_ids = ARRAY(
-            SELECT reviewer_id FROM review_assignment
-            WHERE application_id = NEW.application_id
-        );
-    ELSE
-        reviewer_ids = ARRAY[NEW.reviewer_id];
-    END IF;
-
-    FOREACH rev_id IN ARRAY reviewer_ids LOOP
-        -- Remove existing
-        UPDATE public.application_reviewer_action
-            SET reviewer_action = NULL
-            WHERE application_id = NEW.application_id
-            AND user_id = rev_id;
-
-        -- Insert new
-        WITH reviewer_action AS (
-            SELECT reviewer_action
-                FROM single_application_detail(NEW.application_id, rev_id)
-            ) 
-        INSERT INTO public.application_reviewer_action
-            (user_id, application_id, reviewer_action)  
-        VALUES(
-            rev_id,
-            NEW.application_id,
-            (SELECT reviewer_action FROM reviewer_action)
-         )
-         ON CONFLICT (user_id, application_id)
-         DO UPDATE
-            SET reviewer_action = (SELECT reviewer_action FROM reviewer_action);
-    END LOOP;
-    
-    
-    -- ASSIGNERS
+    -- REVIEWER/ASSIGNER actions
+    -- When updating review_assignments, we re-calculate for every
+    -- reviewer/assigner associated with this application.
```

Comment:
@andreievg -- have updated so that it gets a single list of all reviewers/assigners (from the assigner_join table) and updates them in one pass.